packages/react-admin/src/mui/button/DeleteButton.js

```diff
@@ -29,16 +30,31 @@ const DeleteButton = ({
     translate,
     ...rest
 }) => (
-    <Button
-        className={classnames(classes.link, className)}
-        component={Link}
-        color="accent"
-        to={`${linkToRecord(basePath, record.id)}/delete`}
-        {...rest}
-    >
-        <ActionDelete className={classes.iconPaddingStyle} />
-        {label && translate(label)}
-    </Button>
+    <Responsive
```

Comment:
I'm wondering, why do you use `<Responsive`> at the top level, forcing you to repeat the `<Button>` code, instead of putting it inside the `<Button>`? Is it to avoid introducing a `<span>` to have several children when there is both text and icon?

Reply:
What do you think of:

```jsx
    <Button
        className={classnames(classes.link, className)}
        component={Link}
        color="accent"
        to={`${linkToRecord(basePath, record.id)}/delete`}
        {...rest}
    >
        <ActionDelete />
        <Responsive
            small={<span />}
            medium={<span>{label && translate(label)}</span>}
        />
    </Button>
``` ?

Reply:
much better, but the icon needs a special padding on Mobile

Reply:
No we need a margin between the icon and the label. Solution is to put it on the label